### PR TITLE
Refactoring of the quartz scheduler

### PR DESCRIPTION
**Pull request by klag**

### CODE REVIEW

File: javaServices/coreJavaServices/src/joliex/scheduler/SchedulerService.java [hunk 1]
```diff
@@ -18,158 +19,119 @@
...
+			e.printStackTrace();
 		}
 	}
+	
+	public String getOperationName() {
+		return operationName;
+	}
+	
+	@RequestResponse
+	public Value setCallbackOperation( Value request ) {
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/joliex/scheduler/SchedulerService.java]: Should have void as return type.

File: javaServices/coreJavaServices/src/joliex/scheduler/SchedulerService.java [hunk 2]
```diff
@@ -18,158 +19,119 @@
...
+import static org.quartz.TriggerKey.triggerKey;
+/**
+ *
+ * @author claudio guidi
+ */
+@AndJarDeps({"quartz-2.2.1.jar","quartz-jobs-2.2.1.jar","slf4j-api-1.6.6.jar","c3p0-0.9.1.1.jar"})
 public class SchedulerService extends JavaService
 {
-	public void setDailySchedule( Value request )
+	private Scheduler scheduler;
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/joliex/scheduler/SchedulerService.java]: Can be final.

File: javaServices/coreJavaServices/src/joliex/scheduler/SchedulerServiceJob.java [hunk 3]
```diff
@@ -18,33 +19,41 @@
...
+	@Override
+	public void execute( JobExecutionContext context ) throws JobExecutionException
+	{
+		try {
+			SchedulerContext schedulerContext = context.getScheduler().getContext();
+			SchedulerService service = (SchedulerService) schedulerContext.get( "schedulerService" );
+			Value toSend = Value.create();
+			toSend.getFirstChild( "jobName" ).setValue( context.getJobDetail().getKey().getName() );
+			toSend.getFirstChild( "groupName" ).setValue( context.getJobDetail().getKey().getGroup() );
+			service.sendMessage( CommMessage.createRequest( service.getOperationName(), "/", toSend ));
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/joliex/scheduler/SchedulerServiceJob.java]: The callback operation should be local to the job, not shared among the entire service.

File: javaServices/coreJavaServices/src/joliex/io/FileService.java [hunk 4]
```diff
@@ -434,15 +435,23 @@ private void writeXML(
...
 				try {
 					XSOMParser parser = new XSOMParser();
 					parser.parse( schemaFilename );
 					XSSchemaSet schemaSet = parser.getResult();
-					if ( schemaSet != null ) {
-						type = schemaSet.getElementDecl( "", rootName ).getType();
+					if ( schemaSet != null && schemaSet.getElementDecl( rootNameSpace, rootName ) != null ) {
+						type = schemaSet.getElementDecl( rootNameSpace, rootName ).getType();
+					} else if ( schemaSet.getElementDecl( rootNameSpace, rootName ) == null ) {
+						System.out.println("Root element " + rootName + " with namespace " + rootNameSpace + " not found in the schema " + schemaFilename );
```
Comment (fmontesi) [inline javaServices/coreJavaServices/src/joliex/io/FileService.java]: No println please. Is this an error? Should an exception be thrown?

Comment (klag) [reply javaServices/coreJavaServices/src/joliex/io/FileService.java]: No, it is just a warning message

**Pull request merged**